src/internal/Notification.ts

```diff
@@ -238,41 +238,4 @@ export function observeNotification<T>(notification: ObservableNotification<T>,
   kind === 'N' ? observer.next?.(value!) : kind === 'E' ? observer.error?.(error) : observer.complete?.();
 }
 
-/**
- * A completion object optimized for memory use and created to be the
- * same "shape" as other notifications in v8.
- * @internal
- */
-export const COMPLETE_NOTIFICATION = (() => createNotification('C', undefined, undefined) as CompleteNotification)();
-
-/**
- * Internal use only. Creates an optimized error notification that is the same "shape"
- * as other notifications.
- * @internal
- */
-export function errorNotification(error: any): ErrorNotification {
-  return createNotification('E', undefined, error) as any;
-}
-
-/**
- * Internal use only. Creates an optimized next notification that is the same "shape"
- * as other notifications.
- * @internal
- */
-export function nextNotification<T>(value: T) {
-  return createNotification('N', value, undefined) as NextNotification<T>;
-}
-
-/**
- * Ensures that all notifications created internally have the same "shape" in v8.
- *
- * TODO: This is only exported to support a crazy legacy test in `groupBy`.
- * @internal
- */
-export function createNotification(kind: 'N' | 'E' | 'C', value: any, error: any) {
-  return {
-    kind,
-    value,
-    error,
-  };
-}
+export * from './NotificationFactories';
```

Comment:
Q: Do we need to export this here? Or can we just import these things directly where they are needed?

Reply:
Actually, given that they're marked `@internal`. I think we should not be exporting these from here. We should just import them directly where they are need.

Reply:
It's fine to import them where they're needed. I had to move them because of a circular dependency - that was introduced with their being consumed within `Subscriber` - and I exported them here 'cause this is where they used to live. I can change the import locations instead, if you like.